### ISSUE TITLE
Data Leakage!

### ISSUE THREAD

**Issue by HussainAlasmawi** [floodsung/LearningToCompare_FSL#48]
I found that there is a data leakage in the testing which leads to an increase in the accuracy of the model.

The model contains batch normalization and the batch normalization is supposed to behave differently for training and testing. Since we are testing we suppose to put the model in the model.eval() mode to make the batch normalization behave as in testing but since we didn't add this mode the batch normalization will behave as in the training. it's considered as data leakage and it's increasing the accuracy (for example the Omniglot 5-way 1-shot increases the accuracy from ~90% to 99.6%). Kindly, we need to check this error.
